src/glang/codegen/generatable.py

```diff
@@ -1,18 +1,22 @@
-from typing import Iterable, Optional
+from __future__ import annotations
```

Comment:
Can't we just require a recent python version for this?
Or does this add something else that isn't in 3.10+?

Reply:
There's no version of Python with that feature yet, see https://docs.python.org/3/library/__future__.html#module-contents

It just means that we don't have to wrap type annotations in strings any more

Reply:
Fair. Seems reasonable